docs-src/dev-guide/sdk-features.md

```diff
@@ -0,0 +1,571 @@
+| [Upsert memo](#upsert-memo) | Yes | No   | No         | No     | Yes  | No  |
+
+A Memo is a non-indexed set of Workflow Execution metadata that developers supply at start time or in Workflow code and that is returned when you describe or list Workflow Executions.
+Upsert Memo allows you to update the Memo field for a running Workflow Execution, similar to upserting a Search Attribute.
+This is useful if you want to record additional metadata about the execution as it runs.
+
+<!-- https://temporaltechnologies.slack.com/archives/C01FG4BRQVB/p1696278696796029 -->
+
+### Versioning features
+
+Versioning refers to the ability to update a Workflow Definition without breaking existing Workflow Executions.
+
+| Feature                                             | Go  | Java | TypeScript | Python | .Net | PHP |
+| --------------------------------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Patching APIs](#versioning--patching)              | Yes | Yes  | Yes        | Yes    | Yes  | Yes |
+| [Build ID based dispatch](#build-id-based-dispatch) | Yes | Yes  | No         | Yes    | Yes  | No  |
+
+### Build ID based dispatch
+
+Build ID dispatch is a feature in Temporal that allows running different versions of Workflow and Activity code on the same task queue.
+It works by associating a build ID (typically a version number or git commit hash) with each deployment of Worker code.
+When a Workflow task is dispatched, Temporal will look at the build ID of the Workflow and send the task to a Worker with a compatible build ID.
+This allows seamless deployment of new code versions without Workflow tasks getting stuck.
+Some key aspects of build ID dispatch:
+
+- Workers advertise their build ID to Temporal when polling for tasks.
+- Task Queues maintain sets of compatible build IDs.
+  New IDs can be added to existing compatible sets.
+- When dispatching a Workflow task, Temporal matches the Workflow's build ID to a compatible Worker build ID.
+- Build IDs allow different versions of code to run side by side on the same task queue.
+- Promoting a new build ID to the default set directs new Workflows to start on that code version.
+
+So in summary, build ID dispatch provides a mechanism to deploy new Worker code without disrupting existing Workflows on a Task Queue.
+It's a key feature for zero-downtime deployments in Temporal.
+
+### Patching APIs
+
+Patching APIs refers to the ability to update a Workflow Definition without breaking existing Workflow Executions.
+
+<!--
+Worker Versioning simplifies the process of deploying changes to [Workflow Definitions](/workflows/#workflow-definition).
+It does this by letting you define sets of versions that are compatible with each other, and then assigning a Build ID to the code that defines a Worker.
+The Temporal Server uses the Build ID to determine which versions of a Workflow Definition a Worker can process.
+-->
+
+## Worker-Specific Task Queues
```

Comment:
This is a pattern not a feature. No SDKs have functionality written for this feature.